presage/src/manager.rs

```diff
@@ -953,6 +959,65 @@ impl<C: Store> Manager<C, Registered> {
+                                match state.config_store.contact_by_id(uuid) {
+                                    Ok(Some(_)) => {}
+                                    Ok(None) => {
+                                        // Create a new contact
+                                        if let ContentBody::DataMessage(DataMessage {
+                                            profile_key: Some(profile_key),
+                                            ..
+                                        }) = &content.body
+                                        {
+                                            let contact = Contact {
+                                                uuid: uuid.clone(),
+                                                name: "".to_string(),
+                                                color: None,
+                                                phone_number: None,
+                                                verified: Verified {
+                                                    identity_key: None,
+                                                    state: None,
+                                                    destination_uuid: None,
+                                                    null_message: None,
+                                                },
+                                                blocked: false,
+                                                expire_timer: 0,
+                                                inbox_position: 0,
+                                                archived: false,
+                                                avatar: None,
+                                                profile_key: profile_key.to_vec(),
+                                            };
+                                            // Todo: get the profile from the server
+                                            match state.config_store.save_contact(contact) {
+                                                Ok(_) => {
+                                                    log::info!("Saved contact: {}", uuid);
+                                                }
+                                                Err(e) => {
+                                                    log::error!("Error saving contact: {}", e);
+                                                }
+                                            }
+                                        }
+                                    }
+                                    Err(e) => {
+                                        log::error!("Error getting contact: {}", e);
+                                    }
+                                }
+
+                                let thread = Thread::try_from(&content).unwrap();
+
+                                let store = &mut state.config_store;
+
+                                match store.thread_metadata(&thread) {
```

Comment:
weird that you can't do `state.store.thread_metadata`?

Reply:
You are right i can use the `state.config_store.thread_metadata`.